### ISSUE TITLE
目标检测的实现

### ISSUE THREAD

**Issue by ydzhang12345** [Zehaos/MobileNet#19]
你好，我们刚刚尝试了用你们的model fine tune在了自己的dataset上（traffic sign classification），感觉效果不错。我看到关于目标检测这部分还在debugging之中， 想问下现在主要哪部分还未完善呢？由于时间紧迫，我们需要尽快完成这部分的功能，所以想看看可不可以自己把剩下的部分补充好

**Comment by Zehaos** [comment]
Hi,

目标检测部分已经可以训练，并且是收敛的。

剩下的部分：
1. Evaluation code
2. Optimization
3. 存在的问题： 
  - 当两个gt_bbox匹配到同一个anchor上的时候， 我是直接descard了其中一个gt_bbox。 需要评估下这样会不会导致很大的精度下降。
  - 我本来想追求代码的灵活性， 将label的处理放在graph中，但是现在觉得这样不是很明智，如果将label的处理放在tfrecord中，graph可以很简单（就像squeezedet一样）。
  - lr不能太大， 因为一开始的loss不太稳定，容易Nan。这个问题有可能是第一个问题导致的。

有什么问题可以在这个issue中交流， 谢谢～

**Comment by zsz02** [comment]
我 也是在 训练目标检测， 在VOC_2007上， 总是不能收敛，很困惑 。。。。。

**Comment by Zehaos** [comment]
@zsz02 目前只测试了kitti， 可以把mobilenet搬到squeezeDet中train。

**Comment by kaishijeng** [comment]
I run detection training with Kitti, but loss does not converge ( always stay around 24.00 after 64000 iterations).
Do you see the same behavior?

Thanks,

**Comment by Zehaos** [comment]
Try to tune the learning rate and lr policy. Loss would converge to about 1.5.

**Comment by kaishijeng** [comment]
Can you recommend values of learning rate and lr policy to try?

FC

**Comment by Zehaos** [comment]
You can try:
  --optimizer=sgd
  --learning_rate=0.00005
  --learning_rate_decay_factor=0.5
  --num_epochs_per_decay=10.0

**Comment by davidduo** [comment]
@Zehaos 请问你在检测任务finetune时候，有没有fixed的参数？或者放开了哪些层，固定了哪些层？

**Comment by Zehaos** [comment]
@davidduo 还没尝试固定参数。 一般需要固定前面几层。

**Comment by Goliathplus** [comment]
請問可以用 VOC2007 這類有 multi label 的 data base 來做 classification training 嗎?

**Comment by Zehaos** [comment]
应该是可以的，无非就是把onehot的标签变为'nhot'。VOC有几个类别尺寸比较大，应该可以用。尺寸太小的就不好说。

**Comment by Zehaos** [comment]
@Goliathplus https://arxiv.org/abs/1702.05891 这篇文章看起来不错

**Comment by kaishijeng** [comment]
Google  releases its mobilenet code and pretrained models today (See below). There is a family of mobilenet in this release. Which configuration  is your mobilenet?

https://research.googleblog.com/2017/06/mobilenets-open-source-models-for.html

Thanks,

**Comment by Zehaos** [comment]
@kaishijeng The first one. But still, the others also could be implemented using my code.

**Comment by kaishijeng** [comment]
Thanks for the info.
Do you have a pretrained detection model which you can share? I am still not able to get loss down to below 8 with my own training using values you suggested:
--optimizer=sgd
--learning_rate=0.00005
--learning_rate_decay_factor=0.5
--num_epochs_per_decay=10.0

Thanks,

**Comment by Zehaos** [comment]
@kaishijeng Sorry, no pre-trained detection model is avaliabled now. it is still under developing. And, the most important, I do no guarantee the detection part has no bug.

**Comment by Zehaos** [comment]
Guys,
The official implementation of object detection is now released, see [tensorflow/model/object_detection](https://github.com/tensorflow/models/tree/master/object_detection).

Zehao

**Comment by mydear33000** [comment]
@Zehaos  Detection training with Kitti, loss still can not converge!

**Comment by kaishijeng** [comment]
@Zehaos 

Any idea when you will have a trained detection model to share?

Thanks,
